src/recursion.erl

```diff
@@ -0,0 +1,31 @@
+-module(recursion).
+
+%%
+%% Example:
+%% Different functions to capitalize a string
+%%
+
+%% BAD: makes unnecessary use of manual recursion
+recurse(S) ->
+    lists:reverse(recurse(S, [])).
+
+recurse([], Acc) ->
```

Comment:
Not sure if this is meant as an illustration of your point (if so 👍  😄 ), but this doesn't work.